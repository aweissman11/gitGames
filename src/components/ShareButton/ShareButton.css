.share-btn-box {
  bottom: 40px;
  cursor: pointer;
  position: absolute;
  right: 20px;
  height: 60px;
  width: 160px;
}

.share-btn {
  border-radius: 12px;
  color: #d8daeb;
  font-weight: bold;
  padding: 10px;
  width: 100%;
  background: rgba(220, 220, 220, .4);
}
