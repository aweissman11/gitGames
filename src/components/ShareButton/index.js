import React from 'react';

import './ShareButton.css';

const ShareButton = (props) => {

  const shareThis = async () => {
    const url = `https://cors-anywhere.herokuapp.com/gitgames.herokuapp.com/api/v1/messaging?username=${props.username}`
    const optionsObject = {
      method: "POST", 
    }
    
    const response = await fetch(url, optionsObject)

    if (response.status === 204) {
      alert('email sent to your gitHub followers')
    } else {
      alert('email not sent')
    }
  }

  return (
    <div className='share-btn-box'>
      <p className='share-btn' onClick={() => shareThis()}>Email Followers</p>
    </div>
  )
}

export default ShareButton;