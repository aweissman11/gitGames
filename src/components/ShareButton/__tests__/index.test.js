import React from 'react';
import { shallow } from 'enzyme';

import ShareButton from '../index';

describe('ShareButton', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ShareButton />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
})
