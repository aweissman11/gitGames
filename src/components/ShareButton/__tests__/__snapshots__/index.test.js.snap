// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareButton should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShareButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="share-btn"
        onClick={[Function]}
      >
        Email Followers
      </p>,
      "className": "share-btn-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Email Followers",
        "className": "share-btn",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Email Followers",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="share-btn"
          onClick={[Function]}
        >
          Email Followers
        </p>,
        "className": "share-btn-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email Followers",
          "className": "share-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Email Followers",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
